### PR TITLE
Made logistic regression deterministic for empirical sensitivity analysis

### PR DESCRIPTION
**Pull request by AtheMathmo**
There are some issues with the existing code for the empirical sensitivity analysis.

First, we are using stochastic gradient descent to train the model with a fixed learning rate, meaning we will not actually recover the global optimum. In fact, we will receive a sample within some radius of the global optimum and our sensitivity analysis will be dominated by the variance of this sample.

Second, I found that the default learning rate is too small when using large batch sizes and deterministic training. By increasing it from 0.1 to 10.0 the model trains much faster (it had converged by 100 epochs and gave better loss by 10 epochs compared to lr=0.1).

Finally, in order for the sensitivity of logistic regression to be defined as in Chaudhuri and Monteleon, it must be a deterministic function of the inputs. Thus, given that we are using a relatively small number of epochs and do not guarantee convergence, in addition to making the training dynamics deterministic I found it beneficial to use a fixed random initialization across all models trained. Note that random initialization can still be achieved by changing the model seed between runs.

To understand the need to consider these effects, we can train two nonprivate logistic regression models on the same training distribution and compute the sensitivity. With the existing default settings the sensitivity will be large while with this PR the sensitivity will be 0 as expected.

These changes should improve the sensitivity analysis for 1.3.

Summary of changes:

assignment2/utils.py
- Added option to remove data shuffling

assignment2/empirical_sensitivity.py
- Increase batch size from 256 to 1000
- Make dataloaders have fixed order to allow deterministic training
- Increase learning rate from 0.1 to 10.0 for improved convergence (tested only on larger batchsizes)

assignment2/nonprivate_logistic_regression.py
- Forked the torch RNG (so the RNG state is reset after  model training). This allows us to use the same initialization and makes logistic regression completely deterministic.

### CODE REVIEW

**Comment by ecreager** [comment]
LGTM. nice work
